tests/test_restrictions.py

```diff
@@ -0,0 +1,143 @@
+            [r.effective_date for r in
+             rlm._restrictions_by_asset[self.ASSET1]],
+            [str_to_ts(dt_str) for dt_str in
+             ('2011-01-04', '2011-01-05', '2011-01-06')]
+        )
+
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-03')))
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-03 14:31')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-04')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-04 14:31')))
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-05')))
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-05 14:31')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-06')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-06 14:31')))
+
+        self.assertFalse(rlm.is_restricted(
+            self.ASSET3, str_to_ts('2011-01-03')))
+        self.assertFalse(rlm.is_restricted(
+            self.ASSET3, str_to_ts('2011-01-03 14:31')))
+        self.assertFalse(rlm.is_restricted(
+            self.ASSET3, str_to_ts('2011-01-04')))
+        self.assertFalse(rlm.is_restricted(
+            self.ASSET3, str_to_ts('2011-01-04 14:31')))
+
+        assert_series_equal(
+            rlm.is_restricted([self.ASSET1, self.ASSET2],
+                              str_to_ts('2011-01-03')),
+            pd.Series(data={self.ASSET1: False, self.ASSET2: True})
+        )
+        assert_series_equal(
+            rlm.is_restricted([self.ASSET1, self.ASSET2],
+                              str_to_ts('2011-01-04')),
+            pd.Series(data={self.ASSET1: True, self.ASSET2: True})
+        )
+
+    def test_static_restrictions(self):
+
+        rlm = StaticRestrictions([self.ASSET1, self.ASSET2])
+
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-03')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET2, str_to_ts('2011-01-03')))
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET3, str_to_ts('2011-01-03')))
+
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET1, str_to_ts('2011-01-04')))
+        self.assertTrue(
+            rlm.is_restricted(self.ASSET2, str_to_ts('2011-01-04')))
+        self.assertFalse(
+            rlm.is_restricted(self.ASSET3, str_to_ts('2011-01-04')))
+
+        assert_series_equal(
+            rlm.is_restricted([self.ASSET1, self.ASSET2],
+                              str_to_ts('2011-01-03')),
+            pd.Series(data={self.ASSET1: True, self.ASSET2: True})
+        )
+        assert_series_equal(
+            rlm.is_restricted([self.ASSET1, self.ASSET2],
+                              str_to_ts('2011-01-04')),
+            pd.Series(data={self.ASSET1: True, self.ASSET2: True})
```

Comment:
One thing that would help this test from a readability perspective would be to alias the assets as something like:

```
RESTRICTED_A1 = self.ASSET1
RESTRICTED_A2 = self.ASSET2
UNRESTRICTED = self.ASSET3

`rlm = StaticRestrictions([RESTRICTED_ASSET1, RESTRICTED_ASSET2])`

# Make assertions using the above names.
```